site/releases/2024-jan-22/highlights.qmd

```diff
@@ -0,0 +1,214 @@
+#### `init dataset` improvements
+
+The addition of a `feature_columns` argument allows you to specify a list of feature names for prediction. Internally, the function filters the dataset to retain only these specified features for prediction-related tasks, leaving the remaining dataset available for other purposes, such as segmentation. This improvement replaces the existing behavior of `init_dataset`, which loaded the entire dataset, incorporating all available features for prediction tasks. While this approach worked well, it could impose limitations when generating segmented metrics and proved somewhat inefficient with large datasets containing numerous features, of which only a small subset were relevant for prediction.
+
+Usage example:
+
+```python
+feature_columns = ['CreditScore', 'Age', 'Balance', 'NumOfProducts', 'EstimatedSalary']
+
+vm_train_ds = vm.init_dataset(
+    dataset=train_df,
+    target_column=demo_dataset.target_column,
+    feature_columns=feature_columns
+)
+```
+
+Additionally, two new metrics enable you to report by segment and show the use of `feature_columns`:
+
+- `MissingValuesBySegment`
+- `DescriptiveStatisticsBySegment`
+
+A new notebook illustrates the configuration of dataset features:
+
+- How to utilize the `feature_columns` parameter when initizalizing `validmind` datasets and model objects
+- How `feature_columns` can be used to report by segment
+
+Try it yourself: [Configuring and Using Dataset Features](../../notebooks/how_to/configure_dataset_features.ipynb)
+
+<!---[SC-2707] Support for multiple sections in run_documentation_tests() by @juanmleng in [#307](https://github.com/validmind/developer-framework/pull/307)--->
+#### `run_documentation_tests()` improvements
+
+The `run_documentation_tests()` function, used to collect and run all the tests associated with a template, now supports running multiple sections at a time. This improvement addresses a request we received through customer feedback. The previous behavior was to allow running only one section at a time. This change maintains backward compatibility with the existing syntax, requiring no updates to your code.
+
+Example usage to run one section:
+
+```python
+results = vm.run_documentation_tests(
+    section="data_preparation",
+    dataset=vm_dataset,
+)
+```
+
+Example usage to run multiple sections:
+
+```python
+results = vm.run_documentation_tests(
+    section=["data_preparation", "model_development"],
+    dataset=vm_dataset,
+)
+```
+
+Try it yourself: [Running individual documentation sections](../../notebooks/how_to/run_documentation_sections.ipynb)
+
+<!---feat: [sc-2468] Allow arbitrary test context (V2) by @johnwalz97 in [#312](https://github.com/validmind/developer-framework/pull/312)--->
+#### Support for arbitrary and custom inputs <span id="arbitrary-test-new">
```

Comment:
Can we just say "custom inputs" to simplify the text? That seems like it might entail arbitrary ones unless that is a specific keyword we need to cover.

Reply:
agree

Reply:
@cachafla  what is the difference between "arbitrary" and "custom" inputs?

Reply:
The same. We can just use custom inputs.

Reply:
Updated globally to refer only to "custom inputs".